src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs

```diff
@@ -80,16 +80,30 @@ public string CreateSignedHttpRequest(SignedHttpRequestDescriptor signedHttpRequ
             if (signedHttpRequestDescriptor == null)
                 throw LogHelper.LogArgumentNullException(nameof(signedHttpRequestDescriptor));
 
+            if (signedHttpRequestDescriptor.SigningCredentials == null)
+                throw LogHelper.LogArgumentNullException(nameof(signedHttpRequestDescriptor.SigningCredentials));
+
             var payload = CreateHttpRequestPayload(signedHttpRequestDescriptor, callContext);
+            var header = new JObject();
+            if (signedHttpRequestDescriptor.AdditionalHeaderClaims != null && signedHttpRequestDescriptor.AdditionalHeaderClaims.Count != 0)
+            {
+                if (signedHttpRequestDescriptor.AdditionalHeaderClaims.Keys.Intersect(JwtTokenUtilities.DefaultHeaderParameters, StringComparer.OrdinalIgnoreCase).Any())
+                    throw LogHelper.LogExceptionMessage(new SecurityTokenException(LogHelper.FormatInvariant(JsonWebTokens.LogMessages.IDX14116, nameof(signedHttpRequestDescriptor.AdditionalHeaderClaims), string.Join(", ", JwtTokenUtilities.DefaultHeaderParameters))));
+
+                header.Merge(JObject.FromObject(signedHttpRequestDescriptor.AdditionalHeaderClaims));
+            }
 
-            var additionalHeaderClaims = signedHttpRequestDescriptor.AdditionalHeaderClaims ?? new Dictionary<string, object>();
             // set the "typ" header claim to "pop"
             // https://tools.ietf.org/html/draft-ietf-oauth-signed-http-request-03#section-6.2
-            additionalHeaderClaims[JwtHeaderParameterNames.Typ] = SignedHttpRequestConstants.TokenType;
+            header[JwtHeaderParameterNames.Alg] = signedHttpRequestDescriptor.SigningCredentials.Algorithm;
```

Comment:
> header[JwtHeaderParameterNames.Alg] = signedHttpRequestDescriptor.SigningCredentials.Algorithm; [](start = 11, length = 96)

is this not already specified in the additionalHeaderClaims?

Reply:
?

---
In reply to: [395386176](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1354#discussion_r395386176) [](ancestors = 395386176)